src/ImageSharp/Common/Helpers/ImageMaths.cs

```diff
@@ -147,17 +147,26 @@ public static Rectangle GetBoundingRectangle(Point topLeft, Point bottomRight)
         /// <returns>
         /// The <see cref="Rectangle"/>.
         /// </returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Rectangle GetBoundingRectangle(Rectangle rectangle, Matrix3x2 matrix)
         {
-            var leftTop = Vector2.Transform(new Vector2(rectangle.Left, rectangle.Top), matrix);
-            var rightTop = Vector2.Transform(new Vector2(rectangle.Right, rectangle.Top), matrix);
-            var leftBottom = Vector2.Transform(new Vector2(rectangle.Left, rectangle.Bottom), matrix);
-            var rightBottom = Vector2.Transform(new Vector2(rectangle.Right, rectangle.Bottom), matrix);
-
-            Vector2[] allCorners = { leftTop, rightTop, leftBottom, rightBottom };
-            float extentX = allCorners.Select(v => v.X).Max() - allCorners.Select(v => v.X).Min();
-            float extentY = allCorners.Select(v => v.Y).Max() - allCorners.Select(v => v.Y).Min();
-            return new Rectangle(0, 0, (int)extentX, (int)extentY);
+            // Calculate the position of the four corners in world space by applying
+            // The world matrix to the four corners in object space (0, 0, width, height)
+            var tl = Vector2.Transform(Vector2.Zero, matrix);
+            var tr = Vector2.Transform(new Vector2(rectangle.Width, 0), matrix);
+            var bl = Vector2.Transform(new Vector2(0, rectangle.Height), matrix);
+            var br = Vector2.Transform(new Vector2(rectangle.Width, rectangle.Height), matrix);
+
+            // Find the minimum and maximum "corners" based on the ones above
+            float minX = MathF.Min(tl.X, MathF.Min(tr.X, MathF.Min(bl.X, br.X)));
+            float maxX = MathF.Max(tl.X, MathF.Max(tr.X, MathF.Max(bl.X, br.X)));
+            float minY = MathF.Min(tl.Y, MathF.Min(tr.Y, MathF.Min(bl.Y, br.Y)));
+            float maxY = MathF.Max(tl.Y, MathF.Max(tr.Y, MathF.Max(bl.Y, br.Y)));
+            var min = new Vector2(minX, minY);
+            var max = new Vector2(maxX, maxY);
+            Vector2 size = max - min;
```

Comment:
I wonder if you need the Vector2 here. Why is that better than `float sizeX = maxX -minX`?

Reply:
Tbh, I don't think it is better.